ui/src/components/vault/VaultManagement/History.jsx

```diff
@@ -0,0 +1,190 @@
+import React from 'react';
+import { AmountMath } from '@agoric/ertp';
+import { makeStyles } from '@material-ui/core/styles';
+import Typography from '@material-ui/core/Typography';
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableContainer from '@material-ui/core/TableContainer';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import Paper from '@material-ui/core/Paper';
+
+import { makeDisplayFunctions } from '../../helpers';
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    backgroundColor: '#FFFFFF',
+    marginBottom: theme.spacing(2),
+    borderRadius: '20px',
+    color: '#707070',
+    fontSize: '22px',
+    lineHeight: '27px',
+    padding: theme.spacing(4),
+    minWidth: '50vw',
+  },
+  title: {
+    fontSize: '22px',
+  },
+  table: {
+    minWidth: 650,
+    '& > .MuiTableCell-root': {
+      borderBottom: 'none',
+    },
+  },
+  row: {
+    '& > th, td': {
+      borderBottom: 'none',
+    },
+    '& > th': {
+      paddingLeft: 0,
+    },
+    '& > td': {
+      fontSize: '18px',
+      lineHeight: '24px',
+      color: '#707070',
+    },
+  },
+  left: {
+    paddingLeft: 0,
+  },
+  rowHeader: {
+    '& > *': {
+      fontSize: '16px',
+    },
+  },
+  break: {
+    border: 0,
+    height: '1px',
+    background: '#E5E5E5',
+  },
+  empty: {
+    fontSize: '18px',
+    lineHeight: '24px',
+    color: '#707070',
+    marginTop: theme.spacing(4),
+    marginBottom: theme.spacing(2),
+    textAlign: 'center',
+  },
+}));
+
+const formatDateNow = stamp => {
+  if (!stamp) {
+    return 'unknown time';
+  }
+
+  const date = new Date(stamp);
+  const isoStamp = date.getTime() - date.getTimezoneOffset() * 60 * 1000;
+  const isoDate = new Date(isoStamp);
+  const isoStr = isoDate.toISOString();
+  const match = isoStr.match(/^(.*)T(.*)\..*/);
+
+  return `${match[1]} ${match[2]}`;
+};
+
+const History = ({ vaultId, history, brandToInfo, brand, debtBrand }) => {
+  const { displayAmount, displayBrandPetname } = makeDisplayFunctions(
+    brandToInfo,
+  );
+  const classes = useStyles();
+
+  /**
+   * @typedef  {{
+   * date: number,
+   * deposited: string,
+   * borrowed: string,
+   * id: number
+   * }} Row
+   */
+
+  /**
+   * @type {Row[] | null}
```

Comment:
surprised this type checks since that `&&` expression implies `rows` can be `false`. 

is the type checker not running in CI?

Reply:
Oh hmm, yea it seems like it's not. I get autocomplete in vscode but nothing shouts about a type error.